### PR TITLE
prefer ways on route=bicycle

### LINKED ISSUE #5032: prefer ways with bicycle routes
in bicycle routing, prefer ways that are a part of a `route=bicycle` relation by changing forward/backward rate by a factor of 1.1. Currently it's used by https://github.com/FreemapSlovakia/freemap-routing

Implementation:
**Add this function:**
```(lua)
function get_from_rel(relations, way, key, value, ret)
        -- if any of way's relation have key=value, return tag ret; else return nil
        local rel_id_list = relations:get_relations(way)
        for i, rel_id in ipairs(rel_id_list) do
                local rel = relations:relation(rel_id);
                local p = rel:get_value_by_key(key);
                if value == '*' and p then return rel:get_value_by_key(ret); end
                if p == value then return rel:get_value_by_key(ret); end
        end
        return nil;
end
```
**Create handler**
```(lua)
function WayHandlers.cyclerelation(profile,way,result,data, relations)
    local is_route = get_from_rel(relations, way, "route", 'bicycle', "route");
    if result.forward_rate and is_route then
         result.forward_rate = result.forward_rate *1.1
    end
    if result.backward_rate and is_route then
         result.backward_rate = result.backward_rate *1.1
    end
end
```
and run this handler.

@chaupow @emiltin  what do you think?

### PR DESCRIPTION
**Pull request by MichalPP**
change forward/backward rate on ways, that are a part of route=bicycle. fix #5032

@chaupow @emiltin 

open issues:
* it probably works only if weight_name is cyclability
* fix forward/backward roles (my lua knowledge is too short for this)
* some tests?

### CODE REVIEW

**Comment by MichalPP** [comment]
travis reported errors that it was unable to download some files, otherwise everything passed.

**Comment by MichalPP** [comment]
@TheMarex @chaupow any news on merging?

**Comment by chaupow** [comment]
Looks good! Do you have screenshots of routes that were weird before and are now fixed (for documentation purposes)? 🙂

Also, it would be great if we can find some cucumber tests there are currently failing on master and are fixed with this PR

File: profiles/lib/relations.lua [hunk 1]
```diff
@@ -258,4 +258,18 @@ function Relations.process_way_refs(way, relations, result)
   end
 end
 
+function Relations.filter_relations(relations, way, key, value, ret, forward)
+  -- if any of way's relation have key=value, return tag ret; else return nil
+  -- todo: check backward and forward
```
Comment (emiltin) [inline profiles/lib/relations.lua]: is forward/backward actually handled by this function?

Comment (MichalPP) [reply profiles/lib/relations.lua]: no, my lua knowledge is not that good. (plus I don't know of any such route in my local area).

could you add that check?

**Comment by MichalPP** [comment]
@chaupow @emiltin any chance of merging? my fork is getting too away from the master...

**Comment by emiltin** [comment]
i think forward/backward roles should be handled. other than than i think it would be great to get this merged.

**Comment by DennisOSRM** [comment]
closing stale PR. Reopen if still relevant.

**Pull request closed without merging**